pkg/validation/validation_test.go

```diff
@@ -155,6 +156,185 @@ func TestContainerPorts(t *testing.T) {
+			provider:           "Azure",
+			additionalProvider: "AWS",
+			expected:           false,
+		},
+		{
+			description:        "azure provider with gcp provider parameters specified",
+			provider:           "Azure",
+			additionalProvider: "GCP",
+			expected:           false,
+		},
+		{
+			description:        "gcp provider with aws provider parameters specified",
+			provider:           "GCP",
+			additionalProvider: "AWS",
+			expected:           false,
+		},
+		{
+			description:        "gcp provider with azure provider parameters specified",
+			provider:           "GCP",
+			additionalProvider: "Azure",
+			expected:           false,
+		},
+	}
+
+	name := "test-validation"
+	testString := "projectcontour"
+	cntr := &operatorv1alpha1.Contour{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: fmt.Sprintf("%s-ns", name),
+		},
+		Spec: operatorv1alpha1.ContourSpec{
+			Namespace: operatorv1alpha1.NamespaceSpec{Name: "projectcontour"},
+			NetworkPublishing: operatorv1alpha1.NetworkPublishing{
+				Envoy: operatorv1alpha1.EnvoyNetworkPublishing{
+					Type: operatorv1alpha1.LoadBalancerServicePublishingType,
+					LoadBalancer: operatorv1alpha1.LoadBalancerStrategy{
+						Scope: "External",
+						ProviderParameters: operatorv1alpha1.ProviderLoadBalancerParameters{
+							AWS:   &operatorv1alpha1.AWSLoadBalancerParameters{},
+							Azure: &operatorv1alpha1.AzureLoadBalancerParameters{},
+							GCP:   &operatorv1alpha1.GCPLoadBalancerParameters{},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		if tc.provider == "AWS" {
+			cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type = operatorv1alpha1.AWSLoadBalancerProvider
+			if tc.additionalProvider == "Azure" {
+				cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Azure.Subnet = &testString
+			} else if tc.additionalProvider == "GCP" {
+				cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.GCP.Subnet = &testString
+			}
+		} else if tc.provider == "Azure" {
+			cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type = operatorv1alpha1.AzureLoadBalancerProvider
+			if tc.additionalProvider == "AWS" {
+				cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.AWS.AllocationIDs = strings.Split(testString, "")
+			} else if tc.additionalProvider == "GCP" {
+				cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.GCP.Subnet = &testString
+			}
+		} else if tc.provider == "GCP" {
+			cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type = operatorv1alpha1.GCPLoadBalancerProvider
+			if tc.additionalProvider == "AWS" {
+				cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.AWS.AllocationIDs = strings.Split(testString, "")
+			} else if tc.additionalProvider == "Azure" {
+				cntr.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Azure.Subnet = &testString
+			}
+		}
```

Comment:
nit: This conditional will be easier to read using a switch on the provider parameters type:
```
switch tc.Provider {
  case "AWS":
...
}